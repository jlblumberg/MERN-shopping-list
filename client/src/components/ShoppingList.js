import React, { useState } from 'react';
import { Container, ListGroup, ListGroupItem, Button } from 'reactstrap';
import { CSSTransition, TransitionGroup } from 'react-transition-group';
import { v4 as uuidv4 } from 'uuid';

const ShoppingList = () => {

  const [listItems, setListItems] = useState({
    items: [
      { id: uuidv4(), name: 'Eggs' },
      { id: uuidv4(), name: 'Milk' },
      { id: uuidv4(), name: 'Steak' },
      { id: uuidv4(), name: 'Water' },
    ]
  })

  const onClick = () => {
    const name = prompt('Enter item');
    if (name) {
      setListItems({ items: [...items, { id: uuidv4(), name: name }] })
    }
  }

  return (
    <Container>
      <Button
        id='add-item'
        color='dark'
        style={{ marginBottom: '2rem' }}
        onClick={onClick}
      >Add item
      </Button>
    </Container>
  )

}

export default ShoppingList;