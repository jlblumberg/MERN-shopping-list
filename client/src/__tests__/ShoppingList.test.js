import React from 'react';
import { shallow } from 'enzyme';
import ShoppingList from '../components/ShoppingList';

describe('ShoppingList', () => {
  const wrapper = shallow(<ShoppingList/>);

  it('renders correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('has a button for adding items', () => {
    expect(wrapper.find('#add-item').html()).toContain('Add item');
  });

});