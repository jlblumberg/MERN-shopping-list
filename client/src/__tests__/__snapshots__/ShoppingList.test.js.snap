// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShoppingList renders correctly 1`] = `ShallowWrapper {}`;
